app/src/main/java/org/wikipedia/dataclient/Service.java

```diff
@@ -339,5 +339,19 @@
     Observable<MwPostResponse> postReviewImageLabels(@NonNull @Field("filename") String fileName,
                                                      @NonNull @Field("token") String token,
                                                      @NonNull @Field("batch") String batchLabels);
+    // ------- Watchlist -------
 
+    @GET(MW_API_PREFIX + "action=query&list=watchlist")
+    @NonNull Observable<MwQueryResponse> getWatchlist(@Nullable @Query("continue") String cont);
+
+    @Headers("Cache-Control: no-cache")
+    @POST(MW_API_PREFIX + "action=watch")
+    @FormUrlEncoded
+    Observable<EntityPostResponse> unWatch(@Nullable @Field("unwatch") Integer unwatch,
+                                            @Field("pageids") String pageIds,
```

Comment:
Is it possible to align with the item above?

Reply:
@cooltey done. What do you think about the name? since we use this both for watching and unwatching, should we call it watchUnwatch(...)

Reply:
How about calling it `postWatch` since the `unwatch` is an `Integer` type of parameter and it is also a `post` type of action?

Reply:
since we do not use 'post' method type anywhere else, i will just change it to watch(...) ... does that sound good? and when unwatch is available it is obvious what it does

Reply:
I've noticed that you used `@POST` and also `@FormUrlEncoded`, which is considered as a `post` method type. If that's not your intention, I would suggest you update it to `@GET` and remove the `@FormUrlEncoded`
```
    @POST(MW_API_PREFIX + "action=watch")
    @FormUrlEncoded
```

Reply:
it needs to be POST @cooltey . This is for adding or removing a page from a user's watchlist.

Reply:
> since we do not use 'post' method type anywhere else, i will just change it to watch(...) ... does that sound good? and when unwatch is available it is obvious what it does

Here I meant for naming, we not use post...(...) for any post methods, so i just called it watch(...) instead of postWatch(...)

Reply:
Renamed for consistency.